<launch>
  <arg name="uav_name" default="$(optenv UAV_NAME uav3)"/>

  <arg name="standalone" default="false"/>

  <arg unless="$(arg standalone)" name="nodelet" value="load"/>
  <arg     if="$(arg standalone)" name="nodelet" value="standalone"/>
  <arg unless="$(arg standalone)" name="nodelet_manager" value="$(arg uav_name)_uvdar_nodelet_manager"/>
  <arg     if="$(arg standalone)" name="nodelet_manager" value=""/>

  <!-- Node Settings -->
  <arg name="output" default="screen"/>
  <arg name="proc" default="false"/>
  <arg name="calib" default="false"/>

  <arg name="debug" default="false"/>
  <arg name="gui" default="true"/>

  <arg name="frame_estimate" default="$(arg uav_name)/local_origin_stable"/>
  <arg name="frame_camera" default="color_camera"/>
  <arg name="frame_uvdar" default="uvcam_$(arg uav_name)"/>

  <arg name="offline" default="false"/>

  <arg name="publish_boxes" default="true"/>

  <arg name="calib_file_color" default="calib_results_bf_c_3_6mm.txt"/>

  <!-- <arg name="yaw_diff" default="0.040"/> -->
  <!-- <arg name="pitch_diff" default="-0.01"/> -->
  <!-- <arg name="yaw_diff" default="0.13"/> -->
  <!-- <arg name="pitch_diff" default="0.06"/> -->
  <arg name="yaw_diff" default="0.00"/>
  <arg name="pitch_diff" default="0.00"/>

  <node
    name="uvcam_colorcam_tf"
    pkg="tf"
    type="static_transform_publisher"
    args="0.08 0.00 0.0 0.0 $(arg yaw_diff) $(arg pitch_diff) $(arg frame_uvdar) $(arg frame_camera) 100"/>
    <!-- args="0.08 0.00 0.0 0.0 0.0 0.05 $(arg frame_uvdar) $(arg frame_camera) 100"/> -->
      <!-- <origin xyz="0.108 0 0.085" rpy="1.57079632679 4.71238898038 1.57079632679" /> -->

  <group ns="$(arg uav_name)">

    <node pkg="image_transport" type="republish" name="republish_bluefox" output="screen" args="compressed in:=bluefox/image_raw raw out:=bluefox2/image_raw"  />
                                                  
    <!-- <node pkg="image_transport" type="republish" name="republish_bluefox" output="screen" args="compressed in:=bluefox/image_raw raw out:=bluefox2/image_raw"  /> -->
  
  <!-- <node pkg="uvdar" type="uvdar_kalman_node" name="filterUvdar" output="screen" launch-prefix="debug_roslaunch"> -->
  <!-- <node pkg="uvdar" type="uvdar_reprojector_node" name="reprojector_node" output="screen" launch-prefix="debug_roslaunch"> -->
  <node pkg="uvdar" type="uvdar_reprojector_node" name="uvdar_reprojector_node" output="screen">
    <param name="debug"		type="bool" value="true"/> 
    <param name="gui"             type="bool" value="$(arg gui)"/>

    <param name="frame_camera" type="string" value="$(arg frame_camera)"/>
    <param name="frame_estimate" type="string" value="$(arg frame_estimate)"/>

    <param name="offline" type="bool" value="$(arg offline)"/>

    <param name="calib_file" type="string" value="$(arg calib_file_color)"/>

    <param name="publish_boxes" type="bool" value="$(arg publish_boxes)"/>

    <remap from="measuredPose1" to="/$(arg uav_name)/uvdar/measuredPose1"/>
    <!-- <remap from="measuredPose2" to="/$(arg uav_name)/uvdar/measuredPose2"/> -->
    <remap from="filteredPose1" to="/$(arg uav_name)/uvdar/filteredPose1"/>
    <!-- <remap from="filteredPose2" to="/$(arg uav_name)/uvdar/filteredPose2"/> -->
    <remap from="camera" to="/$(arg uav_name)/bluefox2/image_raw"/>
    <!-- <remap from="camera" to="/$(arg uav_name)/bluefox/image_raw"/> -->
    <remap from="reprojection" to="/$(arg uav_name)/uvdar/reprojection/image_raw"/>
    <remap from="estimatedROIs" to="/$(arg uav_name)/uvdar/estimatedROIs"/>

  </node>
    </group>


  <!-- </group> -->



  </launch>
